src/Features/CSharp/Portable/CodeRefactorings/InlineTemporary/InlineTemporaryCodeRefactoringProvider.ReferenceRewriter.cs

```diff
@@ -12,92 +14,96 @@
+                _conflictReferences = conflictReferences;
+                _nonConflictReferences = nonConflictReferences;
                 _expressionToInline = expressionToInline;
                 _cancellationToken = cancellationToken;
             }
 
-            private bool IsReference(SimpleNameSyntax name)
+            public override SyntaxNode? VisitIdentifierName(IdentifierNameSyntax node)
             {
-                if (name.Identifier.ValueText != _variableDeclarator.Identifier.ValueText)
-                {
-                    return false;
-                }
+                _cancellationToken.ThrowIfCancellationRequested();
 
-                var symbol = _semanticModel.GetSymbolInfo(name).Symbol;
-                return symbol != null
-                    && symbol.Equals(_localSymbol);
+                if (_conflictReferences.Contains(node))
+                    return node.Update(node.Identifier.WithAdditionalAnnotations(CreateConflictAnnotation()));
+
+                if (_nonConflictReferences.Contains(node))
+                    return _expressionToInline;
+
+                return base.VisitIdentifierName(node);
             }
 
-            public override SyntaxNode VisitIdentifierName(IdentifierNameSyntax node)
+            public override SyntaxNode? VisitAnonymousObjectMemberDeclarator(AnonymousObjectMemberDeclaratorSyntax node)
             {
-                _cancellationToken.ThrowIfCancellationRequested();
-
-                if (IsReference(node))
+                if (node.NameEquals == null &&
+                    node.Expression is IdentifierNameSyntax identifier &&
+                    _nonConflictReferences.Contains(identifier))
                 {
-                    if (HasConflict(node, _variableDeclarator))
-                    {
-                        return node.Update(node.Identifier.WithAdditionalAnnotations(CreateConflictAnnotation()));
-                    }
 
-                    return _expressionToInline
-                        .Parenthesize()
-                        .WithAdditionalAnnotations(Formatter.Annotation, Simplifier.Annotation);
+                    // Special case inlining into anonymous types to ensure that we keep property names:
+                    //
+                    // E.g.
+                    //     int x = 42;
+                    //     var a = new { x; };
+                    //
+                    // Should become:
+                    //     var a = new { x = 42; };
+                    return node.Update(SyntaxFactory.NameEquals(identifier), (ExpressionSyntax)Visit(node.Expression));
                 }
 
-                return base.VisitIdentifierName(node);
+                return base.VisitAnonymousObjectMemberDeclarator(node);
             }
 
-            public override SyntaxNode VisitAnonymousObjectMemberDeclarator(AnonymousObjectMemberDeclaratorSyntax node)
+            public override SyntaxNode? VisitArgument(ArgumentSyntax node)
```

Comment:
instead of using the complexifier to add tuple names, we just do it ourselves.  we know we have to as we're replacing a simple name to some arbitrary expression, so we always want to do this.